### DISCUSSION TITLE
How do I extract table from this PDF?

### DISCUSSION THREAD

**Discussion by jakobdo** [jsvine/pdfplumber#445] [Q&A] [answered]
Hello, I am trying the extract the data from this PDF: https://www.nasdaq.com/docs/2020/12/30/Holiday-Calendar-Commodities-Markets.pdf

I have tried with table_settings like this:
table_settings = {"vertical_strategy": "text", "keep_blank_chars": True}

But it skips the last row.
And on page 3 it does "split" the rows after the name of the month.

I have tried using the very nice debug tool: im.reset().debug_tablefinder()
But I am not able to get closer to a working solution. Hope someone can point in the direction to use this library, because it is really nice.

**Comment by samkit-jain** [comment, marked as answer]
Hi @jakobdo Appreciate your interest in the library. I would recommend a 2 step process here. Taking the 3rd page as an example, if you use the `debug_tablefinder()` with the lines strategy (`{"vertical_strategy": "lines", "horizontal_strategy": "lines"}`) you'll notice the output as 
![image](https://user-images.githubusercontent.com/15127115/121167467-9617cf80-c86f-11eb-80ed-e4ad2025bffe.png)

Step 1: Extract the table using the table strategy and store the vertical coordinates as provided by the first row of the table.
```python
tables = page.find_tables()
header_row = tables[0].rows[0].cells
vertical_lines = [cell[0] for cell in header_row] + [header_row[-1][2]]
# Output -> [Decimal('48.625'), Decimal('399.503'), Decimal('684.000')]

```
Step 2: Run table extraction using the explicit vertical lines strategy with the `vertical_lines` as input.
```python
table_settings = {
    "vertical_strategy": "explicit",
    "horizontal_strategy": "lines",
    "explicit_vertical_lines": vertical_lines,  # [Decimal('48.625'), Decimal('399.503'), Decimal('684.000')]
}
```
and the result will be
```
['TRADINGALL WEEK-DAYSEXCEPT ON HOLIDAYS:', 'DATE:']
['New Year', 'January 1']
['Maundy Thursday', '']
['Good Friday', '']
['Easter Monday', '']
['Labor Day', 'May 1']
['Ascension Day', '']
['Whit Monday', '']
['National Day', 'May 17']
['Christmas Eve', 'December 24']
['Christmas Day', 'December 25']
['Boxing Day', 'December 26']
['New Years Eve', 'December 31']
```
![image](https://user-images.githubusercontent.com/15127115/121168979-29053980-c871-11eb-972f-0bb717ddab8d.png)

**Reply by jakobdo** [reply]
Thanks a lot @samkit-jain 
This was very "easy". I will have to play along with this library and this solution. Thanks. :)
